src/dsync/dsync.c

```diff
@@ -3287,6 +3287,22 @@ int main(int argc, char **argv)
             flist_tmp_link = mfu_flist_new();
         }
     }
+    /* Avoid walking if dest doesn't exist or not writable */
+    /* check that destination parent is writable */
+ 
+    /* compute parent path */
+    mfu_path* parent = mfu_path_from_str(destpath->path);
+    mfu_path_dirname(parent);
+    char* parent_str = mfu_path_strdup(parent);
+    mfu_path_delete(&parent);
+
+    /* check that parent is writable */
+    if(mfu_file_access(parent_str, W_OK, mfu_dst_file) < 0) {                                             
```

Comment:
issue: If the user has used the `--dryrun` option, should we skip this check since they will not write to the destination file system at all?

Reply:
shouldn't they hit the same error in dryrun? they can then fix the issue before the real run. Not sure.